### DISCUSSION TITLE
Anyone know how to "define those CSS variables at the component level"?

### DISCUSSION THREAD

**Discussion by ItsRyanWu** [chakra-ui/chakra-ui#6226] [Q&A]
![image](https://user-images.githubusercontent.com/17338101/175013164-7822e4d3-8b12-4525-a536-0b0e5f4798f9.png)

So first I tried using conditional semantic tokens, for instance
```javascript
black: {
  default: '#000000',
  _hover: '#353535'
}
```
And I applied the semantic color 'black' on a element, for example
```html
<Box bg="black" h="10"/>
```
Then when I hover on the page document the background color of the `Box` will change from `#000000` to `#353535`, that's weird but actually expected as the documentation described, because by default the CSS variable is defined on the root element.
So it seems if I defined some CSS variable at the component level then I could avoid the weird behavior and make the background color change only when I hover on the `Box`? Is that what the documentation mean by the sentence "If you are tempted to use a CSS selector like _focus, _hover etc. define those CSS variables at the component level." to avoid the weird behavior? If so how can I do that? I looked up the documentation but found nothing related to it.

**Comment by anubra266** [comment]
You defined `_hover` in your semantic token, so that's the color it's changing to on hover. It's doing exactly what you asked it to do.

**Comment by keithburgie** [comment]
I've been doing a deep dive into conditional tokens and came across this question.

What you're doing with that `black` token _works_, but I think you're right to think it's weird. You probably don't want anything black to change colors any time you hover over it.

I _believe_ that when the documentation refers to the "component level", it is suggesting that you add pseudo selectors like these to [custom component styling](https://chakra-ui.com/docs/styled-system/component-style#styling-single-part-components) or that you apply it to the element directly.

So with your box...
```javascript
// in your theme file
semanticTokens: {
  colors: {
    black: "#000000",
    gray: "#808080"
  }
}

// or...
semanticTokens: {
  colors: {
    black: {
      default: "#000000"
    }
    gray: {
      default: "#808080"
    }
  }
}

// and in your JSX
<Box bg="black" _hover={{ bg: "gray" }}/>
```
